public/app/plugins/datasource/expr/ExpressionQueryEditor.tsx

```diff
@@ -0,0 +1,159 @@
+// Libraries
+import React, { PureComponent, ChangeEvent } from 'react';
+
+import { FormLabel, QueryEditorProps, Select, FormField } from '@grafana/ui';
+import { SelectableValue, ReducerID } from '@grafana/data';
+
+// Types
+import { ExpressionQuery, GELQueryType } from './types';
+import { ExpressionDatasourceApi } from './ExpressionDatasource';
+
+type Props = QueryEditorProps<ExpressionDatasourceApi, ExpressionQuery>;
+
+interface State {}
+
+const gelTypes: Array<SelectableValue<GELQueryType>> = [
+  { value: GELQueryType.math, label: 'Math' },
+  { value: GELQueryType.reduce, label: 'Reduce' },
+  { value: GELQueryType.resample, label: 'Resample' },
+];
+
+const reducerTypes: Array<SelectableValue<string>> = [
```

Comment:
We may want to move all of the reducer,  downsampling and upsampling types to the backend eventually, but no blocker for now. Idea would be that the transform plugin can be queried for meta data about the functions it supports?

Reply:
This is something we should investigate -- it really depends what kind of operations we need and how "custom" the UI needs to get.  it could be something that generates the UI automatically -- or just needs to stay in sync with the frontend.  This would not be a plugin that we "support" extending -- we may have different options in enterprise though